components/buttons/ApplyJob.tsx

```diff
@@ -0,0 +1,28 @@
+import { IJob } from '@/types/job';
+import Button from './Button'
+
+interface IApplyJobButtonProps {
+  job: IJob
+  className?: string
+}
+
+export default function ApplyJobButton({ job,  className = '' }: IApplyJobButtonProps) {
+  const getHref = (contact: string) => {
+    try {
+      new URL(contact);
+    } catch (_) {
+      return `mailto:${contact}`;  
+    }
+
+    return contact;
+  }
+
+  return (
+    <Button
+      href={getHref(job.contact)}
+      target="_blank"
+      text="Apply for this job"
+      className={className}
+    />
+  )
+}
```

Comment:
add a line here to remove this suggestion

Reply:
I have done so @sambhavgupta0705.